python/shotgun_fields/shotgun_field_delegate.py

```diff
@@ -505,7 +505,9 @@ def _set_widget_value(self, widget, model_index):
             primary_item = src_index.model().item(src_index.row(), 0)
             icon = primary_item.icon()
             if icon:
-                widget.set_value(icon.pixmap(QtCore.QSize(256, 256)))
+                # Construct a pixmap object using the available QPixmap class
+                # When running in PySide6, the QPixmap class will be the already patched
+                widget.set_value(QtGui.QPixmap(icon.pixmap(QtCore.QSize(256, 256))))
```

Comment:
I would patch the `QIcon.pixmap` method in tk-core to always create the pixamp from the patched QPixmap class. This way we do not need to remember to convert the native QPixmap to our patched class every time we use QIcon.pixmap